modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/SessionWindowProcessor.java

```diff
@@ -0,0 +1,573 @@
+                        sessionKeyExecutor = (VariableExpressionExecutor) attributeExpressionExecutors[1];
+                    } else {
+                        throw new SiddhiAppValidationException("Session window's session key parameter type"
+                                + " should be string, but found " + attributeExpressionExecutors[1].getReturnType());
+                    }
+                } else {
+                    if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT ||
+                            attributeExpressionExecutors[1].getReturnType() == Attribute.Type.LONG) {
+                        allowedLatency = (Long) ((ConstantExpressionExecutor)
+                                attributeExpressionExecutors[1]).getValue();
+                    } else {
+                        throw new SiddhiAppValidationException("Session window's allowedLatency parameter should be "
+                                + "either int or long, but found " + attributeExpressionExecutors[1].getReturnType());
+                    }
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException("Session window should only have one to three parameters "
+                    + "(<int|long|time> sessionGap, <String> sessionKey, <int|long|time> allowedLatency, "
+                    + "but found " + attributeExpressionExecutors.length + " input attributes");
+
+        }
+
+        if (allowedLatency > sessionGap) {
+            throw new SiddhiAppValidationException("Session window's allowedLatency parameter value "
+                    + "should not be greater than the session gap parameter value");
+
+        }
+
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk,
+                           Processor nextProcessor, StreamEventCloner streamEventCloner) {
+        String key;
+        SessionComplexEventChunk<StreamEvent> currentSession;
+
+        synchronized (this) {
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                long eventTimestamp = streamEvent.getTimestamp();
+                long maxTimestamp = eventTimestamp + sessionGap;
+                long aliveTimestamp = maxTimestamp + allowedLatency;
+
+                if (streamEvent.getType() == StreamEvent.Type.CURRENT) {
+
+                    if (sessionKeyExecutor != null) {
+                        key = (String) sessionKeyExecutor.execute(streamEvent);
+                    } else {
+                        key = DEFAULT_KEY;
+                    }
+
+                    //get the session configuration based on session key
+                    //if the map doesn't contain key, then a new sessionContainer
+                    //object needs to be created.
+                    if (sessionMap.get(key) == null) {
```

Comment:
```suggestion
                    if ((sessionContainer = sessionMap.get(key)) == null) {
```
Change the code as above so that you dont need to call get twice

Reply:
done accordingly